exporter/metric/metric.go

```diff
@@ -643,8 +677,49 @@ func histToDistribution[N int64 | float64](hist metricdata.HistogramDataPoint[N]
 				},
 			},
 		},
-		// TODO: support exemplars
+		Exemplars: toDistributionExemplar[N](hist.Exemplars),
+	}
+}
+
+func expHistToDistribution[N int64 | float64](hist metricdata.ExponentialHistogramDataPoint[N]) *distribution.Distribution {
+	numFiniteBuckets := len(hist.PositiveBucket.Counts)
+	counts := make([]int64, numFiniteBuckets)
+	positiveBucketCounts := 0
+	growthFactor := math.Exp2(math.Exp2(-float64(hist.Scale)))
+	for i, v := range hist.PositiveBucket.Counts {
+		counts[i] = int64(v)
+		positiveBucketCounts += int(v)
 	}
+	var mean float64
+	if !math.IsNaN(float64(hist.Sum)) && positiveBucketCounts > 0 { // Avoid divide-by-zero
+		mean = float64(hist.Sum) / float64(positiveBucketCounts)
+	}
+	return &distribution.Distribution{
+		Count:        int64(positiveBucketCounts),
+		Mean:         mean,
+		BucketCounts: counts,
+		BucketOptions: &distribution.Distribution_BucketOptions{
+			Options: &distribution.Distribution_BucketOptions_ExponentialBuckets{
+				ExponentialBuckets: &distribution.Distribution_BucketOptions_Exponential{
+					Scale:            math.Pow(growthFactor, float64(hist.PositiveBucket.Offset)),
+					GrowthFactor:     growthFactor,
+					NumFiniteBuckets: int32(numFiniteBuckets),
```

Comment:
Looks like the collector exporter also sends an underflow and overflow buckets: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/87124ac8635698cc7cdc9f8f2dfca94a8aa13606/exporter/collector/metrics.go#L1022

Reply:
Fixed. I used as reference the implementation of `collector/metrics.go : exponentialHistogramPoint` to reimplement all the mentioned details of `expHistToDistribution` in this review, but using the correct types and methods.